specs/CookieManagement.md

```diff
@@ -0,0 +1,158 @@
+# Background
+
+Cookie management in WebView has been one of the top feature requests. With that, the WebView2 team has introduced a new set of APIs allowing end developers to achieve goals such as authenticating the webview session, or retrieving cookies from webview to authenticate other tools.
+
+# Description
+
+One can create a `CookieManager` off a WebView to `GetCookies` via a `CookieList` collection, `SetCookie`, `DeleteCookies`, and `ClearAllCookies` in a WebView environment.
+
+# Examples
+
+The following code snippet demonstrates how the cookie management APIs can be use:
+
+```c#
+CoreWebView2CookieManager _cookieManager;
+
+private void WebView_CoreWebView2Ready(object sender, EventArgs e)
+{
+    _cookieManager = webView.CoreWebView2.CreateCookieManager();
+}
+
+void SetCookieCmdExecuted(object target, ExecutedRoutedEventArgs e)
+{
+    CoreWebView2Cookie cookie = _cookieManager.CreateCookie("CookieName", "CookieValue", ".bing.com", "/");

```

Comment:
Is the '.' at the start of .bing.com necessary or have special meaning? The parameter is named 'domain' and I wouldn't offhand expect I need to put a '.' at the start of the parameter.

Reply:
It is necessary for CDP

Reply:
But what does it mean? Does it just mean 'bing.com'? AFAIK the empty domain name label is only valid for the root and is not allowed elsewhere in a domain name. Is this syntax specific to CDP? I don't see anything about this in the [cookie spec](https://tools.ietf.org/html/rfc6265#section-5.1.3). I also don't see anything about it in CDP docs (but that doesn't necessarily mean anything)